extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -0,0 +1,216 @@
+
+export class IntegerValidation extends Validation {
+	constructor(validation: IntegerValidationInfo, valueGetter: ValueGetter) {
+		super(validation, valueGetter);
+	}
+
+	private async isInteger(): Promise<boolean> {
+		const value = await this.getValue();
+		return (typeof value === 'string') ? Number.isInteger(parseFloat(value)) : Number.isInteger(value);
+	}
+
+	getValidator(): Validator {
+		return async () => Promise.resolve({
+			valid: await this.isInteger(),
+			message: this.description
+		});
+	}
+}
+
+export class RegexValidation extends Validation {
+	private _regex: RegExp;
+
+	get regex(): RegExp {
+		return this._regex;
+	}
+
+	constructor(validation: RegexValidationInfo, valueGetter: ValueGetter) {
+		super(validation, valueGetter);
+		throwUnless(validation.regex !== undefined);
+		this._regex = (typeof validation.regex === 'string') ? new RegExp(validation.regex) : validation.regex;
+	}
+
+	getValidator(): Validator {
+		return async () => Promise.resolve({
+			valid: this.regex.test((await this.getValue())?.toString()!),
+			message: this.description
+		});
+	}
+}
+
+export abstract class Comparison extends Validation {
+	private _target: string;
+
+	get target(): string {
+		return this._target;
+	}
+
+	constructor(validation: ComparisonValidationInfo, valueGetter: ValueGetter, variableValueGetter: VariableValueGetter) {
+		super(validation, valueGetter, variableValueGetter);
+		throwUnless(validation.target !== undefined);
+		this._target = validation.target;
+	}
+
+	abstract isComparisonSuccessful(): Promise<boolean>;
+	getValidator(): Validator {
+		return async () => Promise.resolve({
+			valid: await this.isComparisonSuccessful(),
+			message: this.description
+		});
+	}
+}
+
+export class LessThanOrEqualsValidation extends Comparison {
+	async isComparisonSuccessful() {
+		return (await this.getValue())! <= ((await this.getVariableValue(this.target))!);
+	}
+}
+
+export class GreaterThanOrEqualsValidation extends Comparison {
+	async isComparisonSuccessful() {
+		return (await this.getValue())! >= ((await this.getVariableValue(this.target))!);
+	}
+}
+
+
+/**
+ *	removes validation message corresponding to this validator from the @see dialogMessage string.
+ *
+ * @param dialogMessage
+ * @param message
+*/
+export function removeValidationMessage(dialogMessage: azdata.window.DialogMessage, message: string): azdata.window.DialogMessage {
+	if (dialogMessage === undefined) {
+		return dialogMessage;
+	}
+	if (dialogMessage.description) {
+		return {
+			text: dialogMessage.text,
+			level: dialogMessage.level,
+			description: getStrippedMessage(dialogMessage.description, message)
+		};
+	} else {
+		return {
+			text: getStrippedMessage(dialogMessage.text, message),
+			level: dialogMessage.level,
+			description: dialogMessage.description
+		};
+	}
+}
+
+function getStrippedMessage(originalMessage: string, message: string) {
+	if (originalMessage.includes(message)) {
+		const messageWithLineBreak = message + '\n';
```

Comment:
please use EOL instead of '\n' here, you will find the linebreaks not getting removed on windows.

Reply:
do you mind fixing the one in modelviewUtil.ts as well, I have it fixed in my PR, but just realized I abandoned it and created a new PR.

Reply:
yes,  I will do it in my 3rd pr after this is checked in. Look at the draft pr #12945